nf_core/pipelines/bump_version.py

```diff
@@ -161,70 +180,97 @@ def bump_nextflow_version(pipeline_obj: Pipeline, new_version: str) -> None:
             (
                 rf"nextflow%20DSL2-%E2%89%A5{re.escape(current_version)}-23aa62.svg",
                 f"nextflow%20DSL2-%E2%89%A5{new_version}-23aa62.svg",
-            ),
-            (
-                # example: 1. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=20.04.0`)
-                rf"1\.\s*Install\s*\[`Nextflow`\]\(https:\/\/www\.nextflow\.io\/docs\/latest\/getstarted\.html#installation\)\s*\(`>={re.escape(current_version)}`\)",
-                f"1. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>={new_version}`)",
-            ),
+            )
         ],
     )
 
 
-def update_file_version(filename: Union[str, Path], pipeline_obj: Pipeline, patterns: List[Tuple[str, str]]) -> None:
-    """Updates the version number in a requested file.
+def update_file_version(
+    filename: Union[str, Path],
+    pipeline_obj: Pipeline,
+    patterns: List[Tuple[str, str]],
+    required: bool = True,
+    yaml_key: Optional[List[str]] = None,
+) -> None:
+    fn: Path = pipeline_obj._fp(filename)
 
-    Args:
-        filename (str): File to scan.
-        pipeline_obj (nf_core.pipelines.lint.PipelineLint): A PipelineLint object that holds information
-            about the pipeline contents and build files.
-        pattern (str): Regex pattern to apply.
-
-    Raises:
-        ValueError, if the version number cannot be found.
-    """
-    # Load the file
-    fn = pipeline_obj._fp(filename)
-    content = ""
-    try:
-        with open(fn) as fh:
-            content = fh.read()
-    except FileNotFoundError:
+    if not fn.exists():
         log.warning(f"File not found: '{fn}'")
         return
 
-    replacements = []
-    for pattern in patterns:
-        found_match = False
+    if yaml_key:
+        update_yaml_file(fn, patterns, yaml_key, required)
+    else:
+        update_text_file(fn, patterns, required)
```

Comment:
```suggestion
        update_text_file(fn=fn, patterns=patterns, required=required)
```

Reply:
why only for `update_text_file`?

Reply:
Ah sorry I read wrong! I thought it was `update_yaml_file()` and we were missing `yaml_key` 😄 You can ignore my suggestion then